service/history/task/redispatcher.go

```diff
@@ -160,22 +157,21 @@ func (r *redispatcherImpl) AddTask(
 	r.setupTimerLocked()
 }
 
-func (r *redispatcherImpl) Redispatch(
-	targetSize int,
-) {
+func (r *redispatcherImpl) Redispatch(targetSize int) {
 	doneCh := make(chan struct{})
-
-	select {
-	case r.redispatchCh <- redispatchNotification{
+	ntf := redispatchNotification{
 		targetSize: targetSize,
 		doneCh:     doneCh,
-	}:
+	}
+
+	select {
+	case r.redispatchCh <- ntf:
+		// block until the redispatch is done
+		<-doneCh
```

Comment:
was the bug that you found that the done channel would be called during shutdown?

Reply:
not sure if this caused a bug but in previous version of this file line 178 would be called on closed channel during shutdown